### PR TITLE
PRM-2307 New initialise route for estimated LPP

### PR DESCRIPTION
**Pull request by AndrewRButcher**

### CODE REVIEW

File: app/controllers/InitialiseAppealController.scala [hunk 1]
```diff
@@ -70,6 +70,13 @@ class InitialiseAppealController @Inject()(appealService: AppealService,
     }
   }
 
+  def onPageLoadForObligationEstimatedLPP(taxPeriodStartDate: String, taxPeriodEndDate: String): Action[AnyContent] = authorise.async{
+    implicit user =>
+      val appealData = AppealData(PenaltyTypeEnum.Late_Payment, LocalDate.parse(taxPeriodStartDate), LocalDate.parse(taxPeriodEndDate), LocalDate.now(), LocalDate.now())
+      removeExistingKeysFromSessionAndRedirect(
+        routes.CancelVATRegistrationController.onPageLoadForCancelVATRegistration(), "NA", appealData, isAppealAgainstObligation = true)
```
Comment (LewisMitchell) [inline app/controllers/InitialiseAppealController.scala:77]: Just thinking could "NA" be an enum or something more concrete?

File: app/controllers/InitialiseAppealController.scala [hunk 2]
```diff
@@ -70,6 +71,25 @@ class InitialiseAppealController @Inject()(appealService: AppealService,
     }
   }
 
+  def onPageLoadForObligationEstimatedLPP(taxPeriodStartDate: String, taxPeriodEndDate: String): Action[AnyContent] = authorise.async{
```
Comment (LewisMitchell) [inline app/controllers/InitialiseAppealController.scala:74]:
```suggestion
  def onPageLoadForObligationEstimatedLPP(taxPeriodStartDate: String, taxPeriodEndDate: String): Action[AnyContent] = authorise.async {
```

File: it/controllers/InitialiseAppealControllerISpec.scala [hunk 3]
```diff
@@ -173,4 +174,27 @@ class InitialiseAppealControllerISpec extends IntegrationSpecCommonBase {
...
+    "redirect to Cancel VAt Page" in new Setup {
+      implicit val fakeRequest: FakeRequest[AnyContentAsEmpty.type] = FakeRequest().withSession(
+        authToken -> "1234"
+      )
+      val result: Future[Result] = controller.onPageLoadForObligationEstimatedLPP("2025-01-01", "2025-03-31")(fakeRequest)
+      await(result).header.status shouldBe SEE_OTHER
+      redirectLocation(result).get shouldBe routes.CancelVATRegistrationController.onPageLoadForCancelVATRegistration().url
+      val userAnswers = await(userAnswersRepository.collection.find(Document()).toFuture()).head
+      userAnswers.getAnswer[LocalDate](SessionKeys.startDateOfPeriod).isDefined shouldBe true
+      userAnswers.getAnswer[LocalDate](SessionKeys.endDateOfPeriod).isDefined shouldBe true
```
Comment (LewisMitchell) [inline it/controllers/InitialiseAppealControllerISpec.scala:188]: We should have dueDate, penaltyNumber, journeyId and commsDate in the session, could there be a test for that?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: app/controllers/InitialiseAppealController.scala (hunk 2 above)
Comment (AndrewRButcher) [reply app/controllers/InitialiseAppealController.scala]: Fixed ✅

File: it/controllers/InitialiseAppealControllerISpec.scala (hunk 3 above)
Comment (AndrewRButcher) [reply it/controllers/InitialiseAppealControllerISpec.scala:189]: Added ✅

File: app/controllers/InitialiseAppealController.scala [hunk 4]
```diff
@@ -70,6 +71,25 @@ class InitialiseAppealController @Inject()(appealService: AppealService,
     }
   }
 
+  def onPageLoadForObligationEstimatedLPP(taxPeriodStartDate: String, taxPeriodEndDate: String): Action[AnyContent] = authorise.async {
+    implicit user =>
+      try {
+        val penaltyNumberNotApplicable: String = "NA"
```
Comment (LewisMitchell) [inline app/controllers/InitialiseAppealController.scala:77]: May be worth refactoring this at some point

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**